src/sentry/static/sentry/app/views/organizationEventsV2/foo_table.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import {Location} from 'history';
+import {isString, pick} from 'lodash';
+
+import withApi from 'app/utils/withApi';
+import {Client} from 'app/api';
+import {Organization} from 'app/types';
+import {DEFAULT_PER_PAGE} from 'app/constants';
+
+import {EventQuery} from './utils';
+
+type Field = [/* col name */ string, /* field name */ string, /* table width */ number];
```

Comment:
Should we be deserializing into an object instead? Also the widths are going to be automatic for now.

Reply:
@markstory I chose arrays because the resulting encoding is smaller in length:

Object form:

```
field=%7B%22name%22%3A%22title%22%2C%22title%22%3A%22title%22%2C%22width%22%3A50%7D&field=%7B%22name%22%3A%22p95%22%2C%22title%22%3A%22p95%22%2C%22width%22%3A50%7D
```


Array form: 

```
field=%5B%22title%22%2C%22title%22%2C50%5D&field=%5B%22p95%22%2C%22p95%22%2C50%5D
```

Reply:
I meant we could unpack the query data into a class and have that class provide methods that serialize its state back into a query string again. This could reduce our need for utils functions as we can use the object methods.